trigger.tex

```diff
@@ -251,4 +251,16 @@ \section{Trigger Registers}
 \item Write desired value to \RcsrTdataOne.
 \end{steps}
 
+\begin{steps}{Some triggers rely on the effective mode.  The effective mode
+    is the privilege mode that is used by a memory access for purposes of
+    permission checks.  Specifically:}
```

Comment:
There are really two changes here:
1. Clarify "currently active ASID" in textra
2. Mode filtering in mcontrol6

The first one definitely needs to be clarified since, for instance, there's no intuitive definition of "active ASID" in M mode.  My proposal was to have it based on the effective mode (which I think is the right approach) which then led to the second item.

I'm OK backing off on the second item.  I think we'd then naturally want the ASID to be determined by the actual mode rather than the effective mode so the two are consistent.

Something like:
> This trigger will only match if:
> * the mode is VS-mode or VU-mode and ASID in vsatp equals the lower ASIDMAX (defined in the Privileged Spec) bits of svalue.
> * in all other modes, ASID in satp equals the lower ASIDMAX (defined in the Privileged Spec) bits of svalue.

and keeping the commentary:
> Note that sselect and mhselect filtering apply in all modes, including M-mode and S-mode.  If desired, debuggers can use a trigger's mode filtering bits to restrict the matching to modes where it considers ASID/VMID/scontext/hcontext to be active.

Does that sound OK?

Reply:
@pdonahue-ventana, that works for me!